engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/EngineJobExecutionActor.scala

```diff
@@ -479,9 +499,16 @@ class EngineJobExecutionActor(replyTo: ActorRef,
     ()
   }
 
-  private def saveCacheResults(hashes: CallCacheHashes, data: SucceededResponseData) = {
-    callCacheWriteActor ! SaveCallCacheHashes(workflowIdForLogging, hashes, data)
-    val updatedData = data.copy(hashes = Option(Success(hashes)))
+  private def saveCacheResults(hashes: CallCacheHashes, data: CacheWriteResponseData) = {
+    data match {
+      case jobSucceededData: SucceededResponseData =>
+        callCacheWriteActor ! SaveCallCacheHashes(CallCacheHashBundle(workflowIdForLogging, hashes, jobSucceededData.response))
+      case jobFailedNonRetryableData: FailedNonRetryableResponseData =>
+        callCacheWriteActor ! SaveCallCacheHashes(CallCacheHashBundle(workflowIdForLogging, hashes, jobFailedNonRetryableData.response))
```

Comment:
it looks like these two branches are actually the same except for the metadata write on the `FailedNoRetryableResponseData`?